pkg/healthcheck/healthcheck.go

```diff
@@ -2522,3 +2557,101 @@ func containsProxy(pod corev1.Pod) bool {
 	}
 	return false
 }
+
+var (
+	validAsLabelOnly = []string{
+		k8s.DefaultExportedServiceSelector,
+	}
+	validAsAnnotationOnly = []string{
+		k8s.ProxyInjectAnnotation,
+	}
+	validAsAnnotationPrefixOnly = []string{
+		k8s.ProxyConfigAnnotationsPrefix,
+		k8s.ProxyConfigAnnotationsPrefixAlpha,
+	}
+)
+
+func checkMisconfiguredPodsLabels(pods []corev1.Pod) error {
+	var invalid []string
+
+	for _, pod := range pods {
+		invalidLabels := getMisconfiguredLabels(pod.ObjectMeta)
+		if len(invalidLabels) > 0 {
+			invalid = append(invalid,
+				fmt.Sprintf("\t* pod %s/%s contains labels, which should be annotations: %s", pod.Namespace, pod.Name, strings.Join(invalidLabels, ", ")))
+		}
+	}
+
+	if len(invalid) > 0 {
+		return fmt.Errorf("Some labels on pods might be missconfigured:\n%s", strings.Join(invalid, "\n"))
```

Comment:
I was playing around with the messaging around here to be more informative and let users know what exactly is misconfigured. I think this formats the error message in a way that a user can act upon it.

```suggestion
			invalid = append(invalid,
				fmt.Sprintf("\t* %s/%s\n\t\t%s", pod.Namespace, pod.Name, strings.Join(invalidLabels, "\n\t\t")))
		}
	}

	if len(invalid) > 0 {
		return fmt.Errorf("Some labels on data plane pods should be annotations:\n%s", strings.Join(invalid, "\n"))
```

The output looks like this:
```bash
‼ data plane and cli versions match
    emoji-5f947b5dbd-44fkm running git-68cebe88 but cli running dev-undefined
    see https://linkerd.io/checks/#l5d-data-plane-cli-version for hints
‼ data plane pod labels are configured correctly
    Some labels on data plane pods should be annotations:
        * emojivoto/emoji-5f947b5dbd-44fkm
                linkerd.io/inject
        * emojivoto/voting-f7f875999-phcxv
                linkerd.io/inject
                config.linkerd.io/multicluster-option
                config.linkerd.io/viz-option
    see https://linkerd.io/checks/#l5d-data-plane-pod-labels for hints
√ data plane service labels and annotations are configured correctly
```